veloroutes_voies_vertes/actions.py

```diff
@@ -40,3 +40,53 @@ def split_segment(*args):
         iface.messageBar().pushInfo('Véloroutes', msg)
     except QgsProviderConnectionException as e:
         iface.messageBar().pushCritical('Véloroutes', e.args[0])
+
+
+def remove_selected_portion(*args):
+    # get arguments
+    id_iti = int(args[0])
+
+    # get project instance
+    project = QgsProject.instance()
+
+    # get layers needed
+    layerPortions = project.mapLayersByName('v_portion')
+    layerEtape = project.mapLayersByName('etape')
+    pLayer = layerPortions[0]
+    iLayer = layerEtape[0]
+
+    # count number of features selected from v_portion layer
+    count = pLayer.selectedFeatureCount()
+    if count < 1:
+        iface.messageBar().pushCritical('Véloroutes', 'Vous devez selectionner au moins une portion')
+        return
+
+    # get list etape
+    coupleEtape = []
+    for feat in iLayer.getFeatures():
+        if feat['id_itineraire'] == id_iti:
+            coupleEtape.append([feat['id_itineraire'], feat['id_portion']])
```

Comment:
```suggestion
    for feat in iLayer.getFeatures(QgsExpression.createFieldEqualityExpression('id_itineraire', id_iti)):
        coupleEtape.append([feat['id_itineraire'], feat['id_portion']])
```
Utilise une expression ce sera plus rapide